src/app/api/users/route.ts

```diff
@@ -0,0 +1,30 @@
+import { user } from "@/dao/user.dao";
+import { NextResponse, NextRequest } from "next/server";
+import { z } from "zod";
+
+const checkGetAllUsersParams = z.object({
+  username: z.string().optional(),
+  email: z.string().optional(),
+  page: z.number().optional().transform((val) => (val ? Number(val) : undefined)),
+  pageSize: z.number().optional().transform((val) => (val ? Number(val) : undefined)),
+});
+
+export async function GET(req: NextRequest) {
+  try {
+   
+    const queryParams = checkGetAllUsersParams.parse(Object.fromEntries(req.nextUrl.searchParams));
+
+    const { username, email, page, pageSize } = queryParams;
+    
+    const users = await user.fetchUsers({ username, email }, page, pageSize);
+
+    return new NextResponse(JSON.stringify(users), {
+      status: 200,
+    });
```

Comment:
We don't need to convert anything to string, nextjs already does that:

```suggestion
    return Response.json(users)
;
```

Check some examples here:
- https://nextjs.org/docs/app/building-your-application/routing/route-handlers